### ISSUE TITLE
Themes folder read-only

### ISSUE THREAD

**Issue by mr-neal** [nextcloud-snap/nextcloud-snap#915]
Hi

I have a few instances of Nextcloud, all using Snap, some of them have multiple themes depending on the URL so I've been adding theme folders to **snap/nextcloud/current/htdocs/themes** 

However, today I've tried adding a new one and the ftp client gives a general error. So I try creating the directory in terminal using mkdir as root, but I get **mkdir: cannot create directory ‘testdir’: Read-only file system**

This is happening on all instances so I assume it's related to an update. I can see the files I've added previously but I can't add to them or even delete them. Folder permissions are set as 0755

I can add files and directories to other locations right up to snap/nextcloud

Can anyone help me out here?

**Comment by kyrofa** [comment]
`/snap/nextcloud/current/*` has _always_ been read-only. It's a squashfs image; it doesn't even support writing (it's not a permissions issue). I can't quite reason out what you've been doing, can you give me more details?

**Comment by mr-neal** [comment]
Hi, thanks for replying. I think I see where I've become confused... I assumed I'd uploaded the example theme when testing a couple of months ago, but it seems that was there as default.

I'm trying to achieve different themes for different domains on the same installation, as guided here:

https://help.nextcloud.com/t/howto-individual-themes-per-domain/27585

So, if I can't add my own themes then I guess this isn't possible through Snap?

**Comment by sxc731** [comment]
I had the same issue, which I resolved using the following procedure.  I must stress that this is just a workaround:

1. I decided that the theme really belongs in the customizable location for the snap; ie: I uploaded my theme here `/var/snap/nextcloud/current/nextcloud/themes/mytheme`
2. The issue now is indeed that the snap-deployed NC is looking for custom themes in the read-only squashfs image; so I've applied the [usual workaround](https://askubuntu.com/a/1036611/145568); ie: a line similar to this:
`echo -e "/var/snap/nextcloud/current/nextcloud/themes\t/snap/nextcloud/current/htdocs/themes\tnone\tbind,nodev,ro\t0 2" >> /etc/fstab`
3. Now `mount -a` and you're done.

Note that the mount will "hide" the distribution theme directory.  

Ideally I suppose the snap distribution would be modified to look for custom themes in a location similar to the above - assuming this is easily configurable?  An alternative solution would be for NC to allow specifying an absolute path in its `theme` setting.
